core/src/test/java/org/frankframework/jdbc/migration/MigratorTest.java

```diff
@@ -22,90 +24,113 @@
 import org.frankframework.core.BytesResource;
 import org.frankframework.core.Resource;
 import org.frankframework.core.Resource.GlobalScopeProvider;
-import org.frankframework.jdbc.TransactionManagerTestBase;
 import org.frankframework.testutil.ConfigurationMessageEventListener;
+import org.frankframework.testutil.JdbcTestUtil;
 import org.frankframework.testutil.TestAppender;
 import org.frankframework.testutil.TestAssertions;
 import org.frankframework.testutil.TestFileUtils;
+import org.frankframework.testutil.junit.DatabaseTestEnvironment;
+import org.frankframework.testutil.junit.TxManagerTest;
 import org.frankframework.util.AppConstants;
+import org.frankframework.util.JdbcUtil;
 import org.frankframework.util.MessageKeeper;
-import org.junit.Test;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
 
-public class MigratorTest extends TransactionManagerTestBase {
+import lombok.extern.log4j.Log4j2;
 
-	private LiquibaseMigrator migrator = null;
+@Log4j2
+public class MigratorTest {
+
+	private LiquibaseMigrator migrator;
 	private final String tableName = "DUMMYTABLE";
+	private DatabaseTestEnvironment env;
+
+	@BeforeEach
+	public void setup(DatabaseTestEnvironment env) throws Exception {
+		this.env = env;
+
+		migrator = env.createBean(LiquibaseMigrator.class);
+		migrator.setDatasourceName(env.getDataSourceName());
 
-	@Override
-	protected void prepareDatabase() throws Exception {
-		super.prepareDatabase();
-		//Ignore programmatic creation of Temp table, run Liquibase instead!
+		env.getConfiguration().getConfigurationWarnings().destroy();
+		env.getConfiguration().getConfigurationWarnings().afterPropertiesSet();
+		env.getConfiguration().getMessageKeeper().clear();
+	}
+
+	@AfterEach
+	public void teardown() throws Exception {
 		dropTableIfPresent(tableName);
 		dropTableIfPresent("DATABASECHANGELOG");
 		dropTableIfPresent("DATABASECHANGELOGLOCK");
```

Comment:
Misschien specifiek voor deze test een goed idee om ook in de setup deze tabellen te verwijderen, zodat je er altijd zeker van bent dat iedere test begint met een lege database.

En overigens lijkt het me ook een goed idee om _eerst_ via Liquibase zelf te proberen om alle tabellen te droppen, omdat er misschien nog andere tabellen in de database zitten die door Liquibase gemaakt zijn, en die hiermee niet opgeruimd worden (zoals wanneer je bijvoorbeeld op dezelfde lokale database al een keer IAF-Test gedraaid hebt. Nu worden de Liquibase tabellen gedropt maar niet IBISSTORE and IBISTEMP en als je daarna IAF-Test opnieuw draait dan verwacht ik dat je fouten krijgt daarop).

Reply:
de `@WithLiquibase` annotatie gooit alles weg. Hier wordt de migrator (zoals die in het framework zit) getest, en die heeft nog niet eerder dingen gedraaid. Los daarvan zal die nooit zijn eigen tabellen opruimen 😞